searchsummary/src/vespa/searchsummary/docsummary/slime_filler_filter.h

```diff
@@ -0,0 +1,25 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include <vespa/vespalib/stllike/string.h>
+#include <vespa/vespalib/stllike/hash_map.h>
+#include <optional>
+
+namespace search::docsummary {
+
+/*
+ * Class filtering which fields to render in a struct field.
+ */
+class SlimeFillerFilter {
+    vespalib::hash_map<vespalib::string, std::unique_ptr<SlimeFillerFilter>> _filter;
+    std::optional<const SlimeFillerFilter*> get_filter(vespalib::stringref field_name) const;
+public:
+    SlimeFillerFilter();
+    ~SlimeFillerFilter();
+    static std::optional<const SlimeFillerFilter*> get_filter(const SlimeFillerFilter*, vespalib::stringref field_name);
```

Comment:
A comment on how to interpret the result would be nice, to clarify the difference between result.has_value() == true vs false.